### DISCUSSION TITLE
About pkg install

### DISCUSSION THREAD

**Discussion by louies0623** [helloSystem/hello#439] [Ideas]
https://github.com/helloSystem/hello/issues/427#issuecomment-1345243672
https://github.com/helloSystem/hello/issues/423#issuecomment-1344630820
I found that the suggestion you have been giving me is "sudo apk install softwarename" or from FreeBSD Ports, so why don't we build or add a GUI that is dedicated to installing tools, such as "Octopkg", and you have gui pkg installation.

That's what you said.

> Porting software to FreeBSD is generally done in the FreeBSD Ports collection.

https://github.com/helloSystem/hello/issues/264

Octopkg
https://github.com/aarnt/octopkg

**Comment by probonopd** [comment]
FreeBSD uses `pkg` not `apk`.

Using a package manager can be dangerous (requires root, can uninstall other packages including Xorg, etc.), and should hence be only for power users and developers.

For "normal users" we are envisioning a disk image based drag-and-drop "installation" method.

**Reply by louies0623** [reply]
> Using a package manager can be dangerous (requires root, can uninstall other packages including Xorg, etc.), and should hence be only for power users and developers.

Then what the user needs is a pkg installation interface whose UI is a program that hides the system .

> For "normal users" we are envisioning a disk image based drag-and-drop "installation" method.

So what file format should the software use?

**Reply by probonopd** [reply]
Something like `.app` bundles inside `.img` disk images or `.zip` files. Something that doesn't need root rights and can't mess with the operating system.

**Reply by rauljgmesquita** [reply]
Sorry if i'm a bit late, but it's a such good a idea, that apple already uses this on macOS, almost every app you want to install, is distributed as a .dmg (like .img but apple's format) with a .app inside.

**Comment by louies0623** [comment]
Then you can make a similar package manager like this picture, It can not only hide the software built in the system, but also display only the software installed by the user.
![image](https://user-images.githubusercontent.com/44593430/208787486-5233fa9b-9a43-4eff-b98f-ae0473cb800f.png)

**Comment by louies0623** [comment]
I have some ideas. If there is a way to automatically convert the source code into an execution format for BSD.APP without using the Internet, it would save developers the troublesome "conversion", so that more people can support BSD. program.

**Reply by 0x7c28** [reply]
Good idea. But I think this might be difficult to achieve.

**Comment by VincentBentley** [comment]
If the intention is to keep the 'bare metal' install of Hello System clean, you could deploy apps as Podman container images where the application inside the container uses X protocol as a client communicating with X as a display server on Hello System. Podman is available for FreeBSD now and although it doesn't run rootless like on Linux, it creates FreeBSD jails and can run Linux containers in compatibility mode. I guess this is like using Podman as an alternative for Flatpak or Snap. Containers enable the person creating the container to deliver apps that are likely to run on the host system as expected. The hard part will be creating a system that will configure desktop launch icons/menu options on the host system for X applications running in Podman containers and configuring the host firewall (PF) accordingly.
